conn2res/reservoir.py

```diff
@@ -133,33 +165,38 @@ def simulate(self, ext_input, ic=None, threshold=0.5):
         # print('\n GENERATING RESERVOIR STATES ...')
 
         # check data type for ext_input. If list convert to numpy.ndarray
-        if isinstance(ext_input, list): ext_input = np.asarray(ext_input)
+        if isinstance(ext_input, list):
+            ext_input = np.asarray(ext_input)
 
         # initialize reservoir states
-        timesteps = range(1, len(ext_input))
+        timesteps = range(1, len(ext_input)+1)
         self._state = np.zeros((len(timesteps)+1, self.hidden_size))
 
         # set initial conditions
-        if ic is not None: self._state[0,:] = ic
+        if ic is not None:
+            self._state[0, :] = ic
 
         # simulation of the dynamics
         for t in timesteps:
 
             # if (t>0) and (t%100 == 0): print(f'\t ----- timestep = {t}')
+            synap_input = np.dot(
+                self._state[t-1, :], self.w_hh) + np.dot(ext_input[t-1, :], self.w_ih)
+            self._state[t, :] = self.activation_function(synap_input)
 
-            synap_input = np.dot(self._state[t-1,:], self.w_hh) + np.dot(ext_input[t-1,:], self.w_ih)
-            self._state[t,:] = self.activation_function(synap_input)
+        # select output nodes and remove initial condition (to match the time index of
+        # _state and ext_input)
+        self._state = self._state[1:, self.output_nodes]
```

Comment:
These changes together allow to fix the bug that the states of the input nodes were delayed 1 step relative to the input signal.